chainer/functions/math/cholesky.py

```diff
@@ -0,0 +1,64 @@
+import chainer
+from chainer.backends import cuda
+from chainer import function_node
+from chainer.utils import type_check
+from chainer import utils
+
+_cholesky_cpu = None
+
+
+class Cholesky(function_node.FunctionNode):
+
+    @property
+    def label(self):
+        return 'cholesky'
+
+    def check_type_forward(self, in_types):
+        type_check._argname(in_types, ('a'))
+        a_type, = in_types
+
+        type_check.expect(
+            a_type.dtype.kind == 'f'
+        )
+
+    def forward_cpu(self, inputs):
+        a, = inputs
+        global _cholesky_cpu
+        if _cholesky_cpu is None:
+            from numpy import linalg
+            _cholesky_cpu = linalg.cholesky
+        self.retain_outputs((0,))
+        return utils.force_array(_cholesky_cpu(a), dtype=a.dtype),
+
+    def forward_gpu(self, inputs):
+        a, = inputs
+        self.retain_outputs((0,))
+        return utils.force_array(
+            cuda.cupy.linalg.cholesky(a), dtype=a.dtype),
```

Comment:
- Why don't you unify the cpu/gpu implementations?  Define `forward(self, inputs)` and `xp = chainer.backend.get_array_module(a)`.
- `utils.force_array` has no effect because the returned array is 2-dim (not 0-dim) and it has the same dtype.

Reply:
Fixed this as well!